lib/web_ui/lib/src/engine/view_embedder/application_dom/custom_element_application_dom.dart

```diff
@@ -0,0 +1,104 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import '../../dom.dart';
+import 'application_dom.dart';
+
+class CustomElementApplicationDom extends ApplicationDom {
+
+  CustomElementApplicationDom(this._hostElement) {
+    // Clear children...
+    while(_hostElement.firstChild != null) {
+      _hostElement.removeChild(_hostElement.lastChild!);
+    }
+
+    // Hook up a resize observer on the hostElement (if supported!).
+    //
+    // Should all this code live in the DimensionsProvider classes?
+    _resizeObserver = createDomResizeObserver(
+      (List<DomResizeObserverEntry> entries, DomResizeObserver _) {
+        entries.forEach(_streamController.add);
+      }
+    );
+
+    assert(() {
+      if (_resizeObserver == null) {
+        domWindow.console.warn('ResizeObserver API not supported. Flutter will not resize with its hostElement.');
+      }
+      return true;
+    }());
+
+    _resizeObserver?.observe(_hostElement);
+  }
+
+  final DomElement _hostElement;
+  late DomResizeObserver? _resizeObserver;
+
+  final StreamController<DomResizeObserverEntry> _streamController =
```

Comment:
Let's pick a name that reflects this stream controller's role. Otherwise, it just repeast the type name (`String _string` 😛).

Reply:
Damn, my energy to find a good name for a variable must have ran out after writing the ResizeObserver js-interop :P

Reply:
I also moved this to the DimensionsProvider, it makes sense that the same object that helps me compute the size is the one that also tells me that the Size has changed!